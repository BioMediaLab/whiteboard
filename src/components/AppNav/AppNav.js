import React from 'react'
import { Navigation } from '@shopify/polaris'
import { AppUserConsumer } from 'components/AppUser'

export default () => {
  return (
    <AppUserConsumer>
      {props => {
        return (
          <Navigation location="/">
            <Navigation.Section
              items={[
                {
                  url: '/home',
                  label: 'Home',
                  icon: 'home'
                },
                {
                  url: '/classes',
                  label: 'Classes',
                  icon: 'orders'
                }
              ]}
            />
          </Navigation>
        )
      }}
    </AppUserConsumer>
  )
}
