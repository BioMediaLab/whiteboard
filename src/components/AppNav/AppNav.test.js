
/* global render */
import React from 'react'
import AppNav from './AppNav';
import { getByText, fireEvent, cleanup } from '@testing-library/react';


describe("Unit test for AppNav", () => {
    afterEach(cleanup);
    test('AppNav trial', () => {
        const container = render(<AppNav />)
        expect(container.firstChild).toMatchSnapshot();
    })

    test('AppNav get navigation labels', () => {
        const container = render(<AppNav />)
        getByText(container, 'Home');
        getByText(container, 'Courses');
        getByText(container, 'Question Banks');
        getByText(container, 'Media');
        getByText(container, 'Messages');
    })


    test('Home Navigation', () => {
        const container = render(<AppNav />)
        fireEvent.click(getByText(container, 'Home'))
        const elem = getByText(container, 'Home');
        expect(elem).toBeInstanceOf(HTMLSpanElement);
        expect(elem.parentElement).toBeInstanceOf(HTMLAnchorElement);
        expect(elem.parentElement).toHaveAttribute('href', '/home');
    })

    test('Courses Navigation', () => {
        const container = render(<AppNav />)
        const elem = getByText(container, 'Courses');
        expect(elem).toBeInstanceOf(HTMLSpanElement);
        expect(elem.parentElement).toBeInstanceOf(HTMLAnchorElement);
        expect(elem.parentElement).toHaveAttribute('href', '/courses');
    })

    test('Question Bank Navigation', () => {
        const container = render(<AppNav />)
        const elem = getByText(container, 'Question Banks');
        expect(elem).toBeInstanceOf(HTMLSpanElement);
        expect(elem.parentElement).toBeInstanceOf(HTMLAnchorElement);
        expect(elem.parentElement).toHaveAttribute('href', '/question-banks');
    })

    test('Media Navigation', () => {
        const container = render(<AppNav />)
        const elem = getByText(container, 'Media');
        expect(elem).toBeInstanceOf(HTMLSpanElement);
        expect(elem.parentElement).toBeInstanceOf(HTMLAnchorElement);
        expect(elem.parentElement).toHaveAttribute('href', '/media');
    })

    test('Messages Navigation', () => {
        const container = render(<AppNav />)
        const elem = getByText(container, 'Messages');
        expect(elem).toBeInstanceOf(HTMLSpanElement);
        expect(elem.parentElement).toBeInstanceOf(HTMLAnchorElement);
        expect(elem.parentElement).toHaveAttribute('href', '/messages');
    })
    
});