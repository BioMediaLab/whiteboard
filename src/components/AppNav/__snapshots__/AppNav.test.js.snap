// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for AppNav AppNav get link icons 1`] = `
<nav
  class="Polaris-Navigation"
>
  <div
    class="Polaris-Navigation__UserMenu"
  />
  <div
    class="Polaris-Navigation__PrimaryNavigation Polaris-Scrollable Polaris-Scrollable--vertical"
    data-polaris-scrollable="true"
  >
    <ul
      class="Polaris-Navigation__Section"
    >
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/home"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                ui-16px-1_home-52.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Home
          </span>
        </a>
      </li>
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/courses"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                education-16px_school.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Courses
          </span>
        </a>
      </li>
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/question-banks"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                business-16px_bank.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Question Banks
          </span>
        </a>
      </li>
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/media"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                media-16px-1_image-05.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Media
          </span>
        </a>
      </li>
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/messages"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                ui-16px-1_email-83.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Messages
          </span>
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Unit test for AppNav AppNav trial 1`] = `
<nav
  class="Polaris-Navigation"
>
  <div
    class="Polaris-Navigation__UserMenu"
  />
  <div
    class="Polaris-Navigation__PrimaryNavigation Polaris-Scrollable Polaris-Scrollable--vertical"
    data-polaris-scrollable="true"
  >
    <ul
      class="Polaris-Navigation__Section"
    >
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/home"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                ui-16px-1_home-52.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Home
          </span>
        </a>
      </li>
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/courses"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                education-16px_school.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Courses
          </span>
        </a>
      </li>
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/question-banks"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                business-16px_bank.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Question Banks
          </span>
        </a>
      </li>
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/media"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                media-16px-1_image-05.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Media
          </span>
        </a>
      </li>
      <li
        class="Polaris-Navigation__ListItem"
      >
        <a
          class="Polaris-Navigation__Item"
          data-polaris-unstyled="true"
          href="/messages"
          tabindex="0"
        >
          <div
            class="Polaris-Navigation__Icon"
          >
            <span
              class="Polaris-Icon"
            >
              <svg>
                ui-16px-1_email-83.svg
              </svg>
            </span>
          </div>
          <span
            class="Polaris-Navigation__Text"
          >
            Messages
          </span>
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
